clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -289,4 +291,18 @@ public boolean hasExpired() {
     synchronized List<KafkaMetric> metrics() {
         return Collections.unmodifiableList(new LinkedList<>(this.metrics.values()));
     }
+
+    private Object metricLock(Stat stat) {
+        boolean requiresSensorLockForAccess = stat instanceof SampledStat || stat instanceof Rate;
```

Comment:
These pretty much cover all types of metrics that we support. Should we just pass in the Sensor object as the lock all the time?